### ISSUE TITLE
Improving the documentation

### ISSUE THREAD

**Issue by layday** [python/importlib_resources#240]
In `importlib`'s documentation:
 
* Explain what is meant for `ResourceReader` to be "superseded by" `TraversableResources`.
* Document the `TraversableResources` interface (its `files` method).
* Explain that `TraversableResources` is a *concrete* implementation of `ResourceReader`, where only `files` is abstract, not an abstract base class as stated.
* Under the `resources` heading, demarcate the old (soon to be deprecated) API from the new API and discourage usage of the old API.
* Document the `Traversable` interface.  This is a particularly pernicious one because without documentation users do not know which attributes of a `pathlib.Path` object, which is what would be typically returned from `files()`, are safe to use, and which methods are safe to call, with which arguments.

**Comment by jaraco** [comment]
> I am gonna wait for resolution on #80 to do this.

I'd also wait on a call for https://github.com/python/cpython/pull/29036 as that PR is the one that marks the functions as deprecated.

**Comment by jaraco** [comment]
I realize it's a little awkward having issues in this repo and also bpo, _especially_ for `importlib.resources`, whose code lives primarily here but whose docs are exclusively in CPython but where there's a separate set of docs for this project. For the future, please consider filing docs bugs for importlib.resources in [bpo](https://bugs.python.org).

**Comment by jaraco** [comment]
Since some work has been done and this issue seems to be languishing, I'm going to go ahead and close it. That said, I'd welcome anyone to re-invigorate the conversation or provide PRs to improve the situation.
